R/class-plotdata-map-markers.R

```diff
@@ -86,9 +111,17 @@ newMapMarkersPD <- function(.dt = data.table::data.table(),
       attr$binSpec <- binSpec
       attr$binSlider <- binSlider
 
-      xRange <- findViewport(.pd[[x]], xType)
-      .pd[[x]] <- as.character(bin(.pd[[x]], binWidth, xRange))
+      if (is.null(binRange)) {
+        xRange <- findViewport(.pd[[x]], xType)
+      } else {
+        xRange <- validateBinRange(binRange, xType, verbose)
+      }
+      .pd[[x]] <- bin(.pd[[x]], binWidth, xRange, stringsAsFactors=TRUE)
```

Comment:
i may have missed this in a slack discussion, but is there a reason we have numBins defined before taking into account the binRange?

Reply:
Seems like it could differ from the actual total number of bins returned